### PR TITLE
Feature/dlq sns subscription

### PR DESCRIPTION
**Pull request by francilioaraujo**
*Issue #, if available:* #1397

*Description of changes:* Add RedrivePolicy on SNS Event type

*Description of how you validated changes:* Added tests to make sure the RedrivePolicy is passed to CloudFormation template.

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected
- [ ] Add/update example to `examples/2016-10-31`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/model/eventsources/push.py [hunk 1]
```diff
@@ -384,6 +385,9 @@ def to_cloudformation(self, **kwargs):
         if not function:
             raise TypeError("Missing required keyword argument: function")
 
+        if self.RedrivePolicy != None and "deadLetterTargetArn" not in self.RedrivePolicy:
```
Comment (jplock) [inline samtranslator/model/eventsources/push.py]: This should be “is not None”

Comment (francilioaraujo) [reply samtranslator/model/eventsources/push.py]: Thanks. Fixed

**Comment by alexw91** [comment]
# [Codecov](https://codecov.io/gh/awslabs/serverless-application-model/pull/1471?src=pr&el=h1) Report
> Merging [#1471](https://codecov.io/gh/awslabs/serverless-application-model/pull/1471?src=pr&el=desc) into [develop](https://codecov.io/gh/awslabs/serverless-application-model/commit/77fa3b9b027743b2aba62553640496f6231e1cc5&el=desc) will **increase** coverage by `0.03%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/awslabs/serverless-application-model/pull/1471?src=pr&el=tree)

```diff
@@             Coverage Diff             @@
##           develop    #1471      +/-   ##
===========================================
+ Coverage    94.36%   94.40%   +0.03%     
===========================================
  Files           78       78              
  Lines         4685     4718      +33     
  Branches       937      950      +13     
===========================================
+ Hits          4421     4454      +33     
  Misses         121      121              
  Partials       143      143              
```


| [Impacted Files](https://codecov.io/gh/awslabs/serverless-application-model/pull/1471?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [samtranslator/model/sam\_resources.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1471/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9tb2RlbC9zYW1fcmVzb3VyY2VzLnB5) | `94.06% <0.00%> (ø)` | :arrow_up: |
| [samtranslator/model/api/api\_generator.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1471/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9tb2RlbC9hcGkvYXBpX2dlbmVyYXRvci5weQ==) | `95.07% <0.00%> (+0.03%)` | :arrow_up: |
| [samtranslator/model/eventsources/push.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1471/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9tb2RlbC9ldmVudHNvdXJjZXMvcHVzaC5weQ==) | `90.85% <0.00%> (+0.10%)` | :arrow_up: |
| [samtranslator/model/api/http\_api\_generator.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1471/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9tb2RlbC9hcGkvaHR0cF9hcGlfZ2VuZXJhdG9yLnB5) | `98.31% <0.00%> (+0.20%)` | :arrow_up: |
| [samtranslator/open\_api/open\_api.py](https://codecov.io/gh/awslabs/serverless-application-model/pull/1471/diff?src=pr&el=tree#diff-c2FtdHJhbnNsYXRvci9vcGVuX2FwaS9vcGVuX2FwaS5weQ==) | `93.87% <0.00%> (+0.36%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/awslabs/serverless-application-model/pull/1471?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/awslabs/serverless-application-model/pull/1471?src=pr&el=footer). Last update [77fa3b9...3eaf6ea](https://codecov.io/gh/awslabs/serverless-application-model/pull/1471?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by frankleng** [comment]
any update on this? is it just missing docs at this point?

**Comment by jfuss** [comment]
Closing since this has been in draft and not been touched in over a year.

**Pull request closed without merging**